.github/workflows/update-main-source.yml

```diff
@@ -0,0 +1,41 @@
+name: Auto-update charts
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Configure git
+      run: |
+        git config --global user.name "github-actions[bot]"
+        git config --global user.email "github-actions[bot]@users.noreply.github.com"
+
+    - name: Update main-source branch
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_WORKFLOW: ${{ github.workflow }}
+        GITHUB_REPOSITORY: ${{ github.repository }}
+      run: |
+        scripts/pull-ci-scripts
+        BRANCH="auto-update/$(date '+%s')"
+        git checkout -b "$BRANCH"
+        bin/partner-charts-ci auto
+
+        # exit if there are no changes
+        git diff --quiet main-source "$BRANCH" && exit 0
+
+        # close all existing PRs from branches starting with "auto-update"
+        gh pr --repo "$GITHUB_REPOSITORY" list --search 'head:auto-update' --json 'headRefName' --jq '.[] | join("\n")' | \
```

Comment:
One possible issue is that you should always make sure that this Ubuntu latest image has JQ installed; otherwise, it will break. 
My Ubuntu did not come with it installed.

Reply:
I don't think this command requires `jq` - it's just an option. Looking at the dependencies of the `gh` package, only `git` is listed. So they must build `jq`, or its syntax, into `gh` somehow?